docs/Ros2Extension.md

```diff
@@ -0,0 +1,110 @@
+4. Copy all ROS2 libraries from `<build-dir>/ros2_standalone/` to the newly created  `.plugin` directory.
+
+This way, ROS2 standalone builds for `ROS2ForUnity` and `RGL` will be separated and not loaded twice by Unity.
+
+### Windows
+
+#### Prerequisites
+
+- Requirements listed in the main [README](../README.md) (section `Building on Windows`).
+- ROS2 installed on the system and sourced.
+- Fixed ROS2 logging macros in rclcpp package to make it compile with C++20. More about this bug: [github PR](https://github.com/ros2/rclcpp/pull/2063).
+  - Use `fix_ros2_humble.py` script to apply those changes:
+    ```bash
+    py ros2_standalone\fix_ros2_humble.py <your-path-to-ros2>
+    ```
+- For standalone build:
+  - If you have installed ROS2 with binaries, there is `rmw_cyclonedds_cpp` package missing. You need to build this one from source:
+    1. Make sure you have installed [ROS2 prerequisites](https://docs.ros.org/en/humble/Installation/Alternatives/Windows-Development-Setup.html#installing-prerequisites).
+    2. Good practice is to disable Windows path limits, [see](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry).
+    3. Run `x64 Native Tools Command Prompt for VS 2019`, setup development folder (let say: `C:\rosrmw`), clone ROS2 repos, and build:
+        ```bash
+        md \rosrmw\src
+        cd \rosrmw
+        vcs import --input https://raw.githubusercontent.com/ros2/ros2/humble/ros2.repos src
+        colcon build --merge-install --packages-up-to rmw_cyclonedds_cpp
+        ``` 
+    4. Before building RGL, source `rosrmw` workspace: `call C:\rosrmw\install\setup.bat`.
+
+
+#### Steps for overlay
+1. Run `x64 Native Tools Command Prompt for VS 2019` and navigate to RGL repository.
+2. Run `python setup.py --with-ros2` command to build RGL with ROS2 extension.
+
+#### Steps for standalone
+1. Run `x64 Native Tools Command Prompt for VS 2019` and navigate to RGL repository.
+2. Run `python setup.py --with-ros2-standalone` command to build RGL with ROS2 extension and install ROS2 libraries.
+3. Copy all ROS2 libraries from `<build-dir>/ros2_standalone/` into `RobotecGPULidar.dll` location, or extend environment variable `Path` appropriately.
+
+#### Tips for integrating RGL + ROS2 standalone with Unity and [ROS2ForUnity](https://github.com/RobotecAI/ros2-for-unity) plugin.
+1. Build RGL with ROS2 standalone as described above.
+2. Copy `<build-dir>/RobotecGPULidar.dll` and all depend libraries located in `<build-dir>` to appropriate directory in your RGLUnityPlugin.
+3. Copy all ROS2 libraries from `<build-dir>/ros2_standalone/` to `Ros2ForUnity\Plugins\Windows\x86_64` directory. Skip for duplicates.
+
+In this case, RGL's ROS2 standalone build is dependent on ROS2ForUnity's ROS2 standalone build. RobotecGPULidar library will find ROS2 because ROS2ForUnity sets environment variable `Path` for the ROS2 libraries.
```

Comment:
Does it mean that RGL is always loaded after R2FU? How is the order handled?

Reply:
R2FU is initialized before any `Awake()` or `Start()`. This is achieved by [RuntimeInitializeOnLoadMethod](https://docs.unity3d.com/ScriptReference/RuntimeInitializeOnLoadMethodAttribute.html) attribute. See [source code](https://github.com/tier4/AWSIM/blob/main/Assets/AWSIM/Scripts/ROS/SimulatorROS2Node.cs#L20).